src/bramble/client/StateManager.js

```diff
@@ -0,0 +1,106 @@
+/*global define*/
+/*jslint bitwise: true */
+define(function() {
+    "use strict";    
+
+    var memoryStorage = {
+        _items: {},
+        getItem: function(key) {
+            return memoryStorage._items[key];
+        },
+        setItem: function(key, value) {
+            // Mimic localStorage string storage
+            value = "" + value;           
+            memoryStorage._items[key] = value;
+        },
+        clear: function() {
+            memoryStorage._items = {};
+        }
+    };
+
+    var localStorage = (function(window) {
+        if(typeof window.localStorage === 'undefined') {
+            return memoryStorage;
+        }
+        return window.localStorage;
+    }(window));
+
+    function prefix(name) {
+        return "bramble-property::" + name;
+    }
+
+    function getString(storage, property) {
+        var val = storage.getItem(prefix(property));
+        return val || null;
```

Comment:
It just makes what `localStorage.getItem()` does when the key isn't there more explicit (i.e., it returns null, but this documents the fact that `getString` can return null).